icu4c/source/test/intltest/units_test.cpp

```diff
@@ -708,10 +708,22 @@ void UnitsTest::testComplexUnitsConverterSorting() {
 
     for (const auto &testCase : testCases) {
         MeasureUnitImpl inputImpl = MeasureUnitImpl::forIdentifier(testCase.input, status);
+        if (U_FAILURE(status)) {
+          return;
+        }
         MeasureUnitImpl outputImpl = MeasureUnitImpl::forIdentifier(testCase.output, status);
+        if (U_FAILURE(status)) {
+          return;
```

Comment:
```suggestion
            return;
```

Reply:
Better:

```suggestion
        if (status.errIfFailureAndReset()) {
            continue;
        }
```

Reply:
+1

Reply:
(Redid my suggestions as I did them wrong - couldn't be directly applied. Deleted the old ones.)